feat: functionality improvements to util script

deborahgu opened the pull request:
pull-all-repos-in-directory enhancements:
* -q flag to make it less noisy
* -c flag so that it will do garbage collection and committed branch cleanup
* -d flag  so you can specify a top-level directory to run in

without any of these flags, behavior should be unchanged.

----

I've completed each of the following or determined they are not applicable:

- [x] Made a plan to communicate any major developer interface changes (or N/A)

clintonb commented:
@edx/docker-devstack-working-group please review

jbarciauskas commented:
I'd rather not take this shortcut. I added set -e where I did specifically because the database loading process was failing silently. It turned out I had a half-loaded database and needed to reset things.

clintonb commented:
@jbarciauskas in that case can you prioritize fixing this or documenting a workaround? Folks no longer know what to do when they update code, and I no longer know what to tell them. The system is essentially unusable unless people completely destroy their devstacks and provision from scratch. In some cases where data setup is a tedious, manual process this is an unacceptable demand.

jbarciauskas commented:
I don't understand - the update process is the same as it's always been? Switch branches, then run migrations and run asset compilation right? Is there more?

clintonb commented:
That's easily done on one IDA. What if I want to provision all of them?  It's worth asking: what is the usecase for provisioning? Today with Vagrant it is used to update all services. Right now it's a one-time command for Docker devstack. I find that rather confusing.

robrap commented:
@clintonb FYI: I don't know anyone who provisions Vagrant devstack other than the initial provisioning.  I think the process we all followed in old TNL was to get a new devstack if there was a major change that was needed.

jbarciauskas commented:
Yeah, my understanding of the use case was just things go foobar and you want to start over because it seems easier than sorting out what went wrong. For incremental updates like branch switches, wasn't there an update-lms make target under discussion?

clintonb commented:
Ah! There lies the confusion. The provision target, as I originally envisioned, was meant to be run anytime after a code update. The idea was that you could update your code and run one commeand to get all of the services functions properly (e.g. dependency installation, migrations, assets).

I do not recall other discussions on provisioning.

jmbowman commented:
I don't think we're hitting errors in this stage anymore, people have been re-provisioning the environment successfully for a while now.

deborahgu committed:
feat: functionality improvements to util script

pull-all-repos-in-directory enhancements:
* -q flag to make it less noisy
* -c flag so that it will do garbage collection and committed branch cleanup
* -d flag  so you can specify a top-level directory to run in

without any of these flags, behavior should be unchanged.

justinhynes reviewed (approved):
FWIW -- I'm not very adept at shell scripting, but I pulled and tested it and seems good! It looks like it did what was desired/expected and I like the cleanup options.

One nit -- I had to `chmod +x` the script to make it executable once I pulled down your branch. Can we update the permissions before merge?

deborahgu committed:
chore: updating file permissions

JH  pointed out that the file permissions should be 755.
